datalad/tests/utils_pytest.py

```diff
@@ -1632,57 +1632,72 @@ def ignore_nose_capturing_stdout(func):
         OBSCURE_FILENAME_PARTS.append(UNICODE_FILENAME)
 # space before extension, simple extension and trailing space to finish it up
 OBSCURE_FILENAME_PARTS += [' ', '.datc', ' ']
+# Parts that should not be used in directory names
+DIRECTORY_EXCLUDED_PARTS = ['\n']
 
 
 @with_tempfile(mkdir=True)
-def get_most_obscure_supported_name(tdir, return_candidates=False):
+def get_most_obscure_supported_name(tdir, return_candidates=False, directory_name=False):
     """Return the most obscure filename that the filesystem would support under TEMPDIR
 
     Parameters
     ----------
     return_candidates: bool, optional
       if True, return a tuple of (good, candidates) where candidates are "partially"
       sorted from trickiest considered
+
+    directory_name: bool, optional
+      if True, return an obscure directory name. Directories might have
+      different obscurity profiles. Currently, directory names will not
+      contain newline characters.
```

Comment:
Why so definitive? It is that we don't support new lines in directory - if so, please add a note why

Reply:
Yes, it is that we do not support new lines in directories. I commented in detail in issue #7173 in https://github.com/datalad/datalad/issues/7173#issuecomment-1314968568. Basically `git-annex` does not work in annex repositories that have a newline in the name of the root directory. Will add a comment.

Reply:
Comment added

Reply:
Thanks! Might be worth alerting @joeyh to that fact as well

Reply:
filed https://git-annex.branchable.com/bugs/init_fails_in_a_folder_with_newline_in_its_name/?updated .
Overall I think that
- git-annex issue should be fixed
- we should not bother distinguishing obscure path from folder -- our detection here doesn't help for git-annex case, and otherwise we seems to have no other use case where obscure filename cannot be obscure directory name *given filesystem limitations* and not bugs in the software, right?

Reply:
> 

@yarikoptic : using newline in directories is a snake-pit. Try, for example, the following in a datalad dataset:
```bash
> datalad create -d . "abc
def"
```

That leads to an unusable dataset because ".git/config" and ".gitmodules" become invalid (even adding a submodule with newline in git, does leave the repo in an inconsistent state, because a .gitmodule entry for a commit-object is missing). 


Using OBSCURE_FILENAME for directories introduces a problem with submodule configuration entries in
`.git/config` and `.gitmodules`.  For example, a submodule called "abc\ndef", i.e. "abc" plus a newlines plus "def", leads to the following invalid entry in `.git/config`:

```
[submodule "abc
def"]
        active = true
        url = ./abc
def
```

and the following incorrect entry in `.gitmodules`.

```
[submodule "abc
def"]
        path = abc
def
        url = ./abc
def
        datalad-id = d9f3edc7-a39d-44db-9e74-4bb37741e08d
```

To allow newlines in subdataset names, we would have to quote the submodule name requires quoting of submodule names, paths and URLs. The folliowing `.gitmodules` and `.git/config` entries would work

.gitmodules:
``` 
[submodule "abc\ndef"]
        path = "abc\ndef"
        url = "./abc\ndef"
        datalad-id = d9f3edc7-a39d-44db-9e74-4bb37741e08d
``` 

.git/config
``` 
[submodule "abc\ndef"]
        active = true
        url = "./abc\ndef"
``` 

But not even git creates the proper entries, when adding a submodule with a newline in the directory name.

So in conclusion, we should not try to use newlines in directories

Reply:
Good digging.  I think it is 
- worth indeed just not bother about `\n` in directories for now
- report finding (allowed by filesystem and `git` per se - but doesn't work correctly) to `git` folks